src/modules/commander/Commander.cpp

```diff
@@ -1797,6 +1798,15 @@ Commander::run()
 			}
 		}
 
+		const bool just_disarmed = !armed.armed && was_armed;
+		const bool just_finished_auto_mission = is_auto_state(internal_state.main_state) && _mission_result_sub.get().finished;
+		const bool last_state_valid = last_non_auto_state != commander_state_s::MAIN_STATE_MAX;
+
+		if(just_disarmed && land_detector.landed && just_finished_auto_mission && last_state_valid){
+			PX4_INFO("Just finished auto mission, transitioning back to last manual mode.");
```

Comment:
I'm still struggling to assess if this should not really just be limited to RTL instead of all auto states. What is the use case wanting to return from mission to manual without RTL? What if I'm only steering the vehicle with a tablet?

Reply:
I'm not sure. This change was originally requested by @MikeGoesDrones , so maybe he can weigh in on his rationale.

Reply:
the RTL was the most annoying one to deal with. i don't see a use case in mission right now. either you land manually or with RTL.

Reply:
Ok, then will change it to just work with RTL.